fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccountingProcessorHelper.java

```diff
@@ -909,12 +911,13 @@ private void createDebitJournalEntryForSavings(final Office office, final String
         LoanTransaction loanTransaction = null;
         SavingsAccountTransaction savingsAccountTransaction = null;
         ClientTransaction clientTransaction = null;
-        final PaymentDetail paymentDetail = null;
+        PaymentDetail paymentDetail = null;
         final Long shareTransactionId = null;
         String modifiedTransactionId = transactionId;
         if (StringUtils.isNumeric(transactionId)) {
             long id = Long.parseLong(transactionId);
             savingsAccountTransaction = this.savingsAccountTransactionRepository.findOne(id);
+            paymentDetail = this.savingsAccountTransactionRepository.findOne(id).getPaymentDetail();
```

Comment:
unnecessary repository call. You can make use of savingsAccountTransaction object to retrieve payment details